### DISCUSSION TITLE
Convert C Array to BDF or other format

### DISCUSSION THREAD

**Discussion by Galaxy-Man** [Bodmer/TFT_eSPI#2692] [Q&A - Fonts] [answered]
Hi 

Has anyone got a utility etc to take an arry from fonts and convert to a BDF or other format for editing?

Thank you for any help.

example font

const uint8_t Futura_Heavy_font4pt7bBitmaps[] PROGMEM = {
  0x00, 0xF4, 0xB4, 0x38, 0xA7, 0xCC, 0xF9, 0xC0, 0xFD, 0x1C, 0x5A, 0xF8,
  0x80, 0xFA, 0xCE, 0x0F, 0x35, 0x70, 0x72, 0x99, 0xF9, 0x7C, 0xC0, 0x6A,
  0xA4, 0x95, 0x58, 0xFF, 0x80, 0x21, 0x3E, 0x42, 0x00, 0x60, 0xC0, 0x80,
  0x12, 0x24, 0x48, 0x69, 0x99, 0x96, 0xD5, 0x50, 0x79, 0x12, 0x6F, 0xE1,
  0x12, 0x97, 0x11, 0x94, 0xAF, 0x88, 0x74, 0xE1, 0x1E, 0x64, 0xE9, 0x96,
  0xF2, 0x26, 0x4C, 0x69, 0x96, 0x9F, 0x69, 0x97, 0x26, 0x90, 0x41, 0x80,
  0x1F, 0x38, 0x30, 0xF8, 0x3E, 0xC1, 0xCF, 0x80, 0xF5, 0xA0, 0x80, 0x3C,
  0x8E, 0x6D, 0x5B, 0xD8, 0x9E, 0x00, 0x30, 0xC7, 0x92, 0x7B, 0x30, 0xF9,
  0xE9, 0x9E, 0x7E, 0x61, 0x0C, 0xBC, 0xF4, 0xE3, 0x18, 0xF8, 0xF3, 0xC9,
  0xC0, 0xF3, 0xC9, 0x00, 0x7B, 0x18, 0x23, 0xC5, 0xE0, 0x8C, 0x7F, 0x18,
  0xC4, 0xFC, 0x24, 0x9B, 0xC0, 0x9D, 0xB9, 0xCB, 0x4C, 0x92, 0x49, 0xC0,
  0x44, 0xD9, 0xB5, 0x59, 0x32, 0x40, 0xCE, 0x7B, 0x79, 0xC4, 0x7B, 0x38,
  0x61, 0xCD, 0xE0, 0xF9, 0xF8, 0x88, 0x7B, 0x38, 0x61, 0xDD, 0xE0, 0x40,
  0xF9, 0xEC, 0xA9, 0xF8, 0x61, 0x9E, 0xE9, 0x24, 0x80, 0x8C, 0x63, 0x18,
  0xB8, 0x8E, 0xD4, 0xA5, 0x10, 0x99, 0xBD, 0xBD, 0xA6, 0x66, 0x42, 0xDA,
  0x88, 0xE5, 0x6C, 0xDA, 0x88, 0x42, 0x10, 0xF8, 0x88, 0x44, 0x7C, 0xEA,
  0xAC, 0x84, 0x42, 0x21, 0xD5, 0x5C, 0x76, 0xC0, 0xF0, 0xC0, 0xF9, 0x97,
  0x88, 0xF9, 0x9E, 0x72, 0x30, 0x11, 0xF9, 0x97, 0x7F, 0x87, 0xEE, 0xA0,
  0xF9, 0x97, 0x1F, 0x88, 0xF9, 0x99, 0xBC, 0xBF, 0x88, 0xBE, 0xEB, 0xFC,
  0xFF, 0x26, 0x4C, 0x90, 0xF9, 0x99, 0x69, 0x96, 0xF9, 0x9E, 0x88, 0xF9,
  0x97, 0x11, 0xF2, 0x40, 0xF1, 0xF0, 0x5D, 0x24, 0x99, 0x9F, 0xDA, 0x94,
  0x40, 0xB6, 0xD7, 0x92, 0xF6, 0x6F, 0xD7, 0x62, 0x64, 0xF2, 0x4F, 0x69,
  0x44, 0x98, 0xFF, 0xC9, 0x14, 0xB0, 0xEC, 0xC0 };

const GFXglyph Futura_Heavy_font4pt7bGlyphs[] PROGMEM = {
  {     0,   1,   1,   2,    0,    0 },   // 0x20 ' '
  {     1,   1,   6,   3,    1,   -5 },   // 0x21 '!'
  {     2,   3,   2,   3,    0,   -4 },   // 0x22 '"'
  {     3,   6,   6,   6,    0,   -5 },   // 0x23 '#'
  {     8,   5,   7,   5,    0,   -5 },   // 0x24 '$'
  {    13,   6,   6,   7,    0,   -5 },   // 0x25 '%'
  {    18,   5,   6,   6,    1,   -5 },   // 0x26 '&'
  {    22,   1,   2,   1,    0,   -4 },   // 0x27 '''
  {    23,   2,   7,   3,    1,   -5 },   // 0x28 '('
  {    25,   2,   7,   3,    1,   -5 },   // 0x29 ')'
  {    27,   3,   3,   4,    1,   -5 },   // 0x2A '*'
  {    29,   5,   5,   7,    1,   -4 },   // 0x2B '+'
  {    33,   2,   2,   2,    0,    0 },   // 0x2C ','
  {    34,   2,   1,   3,    1,   -1 },   // 0x2D '-'
  {    35,   1,   1,   2,    1,    0 },   // 0x2E '.'
  {    36,   4,   6,   4,    0,   -5 },   // 0x2F '/'
  {    39,   4,   6,   5,    1,   -5 },   // 0x30 '0'
  {    42,   2,   6,   5,    1,   -5 },   // 0x31 '1'
  {    44,   4,   6,   5,    1,   -5 },   // 0x32 '2'
  {    47,   4,   6,   5,    1,   -5 },   // 0x33 '3'
  {    50,   5,   6,   5,    0,   -5 },   // 0x34 '4'
  {    54,   4,   6,   5,    1,   -5 },   // 0x35 '5'
  {    57,   4,   6,   5,    1,   -5 },   // 0x36 '6'
  {    60,   4,   6,   5,    0,   -5 },   // 0x37 '7'
  {    63,   4,   6,   5,    1,   -5 },   // 0x38 '8'
  {    66,   4,   6,   5,    1,   -5 },   // 0x39 '9'
  {    69,   1,   4,   3,    1,   -3 },   // 0x3A ':'
  {    70,   2,   5,   3,    0,   -3 },   // 0x3B ';'
  {    72,   5,   4,   7,    1,   -3 },   // 0x3C '<'
  {    75,   5,   3,   7,    1,   -3 },   // 0x3D '='
  {    77,   5,   4,   7,    1,   -3 },   // 0x3E '>'
  {    80,   3,   6,   5,    1,   -5 },   // 0x3F '?'
  {    83,   7,   7,   8,    0,   -5 },   // 0x40 '@'
  {    90,   6,   6,   6,    0,   -5 },   // 0x41 'A'
  {    95,   4,   6,   5,    1,   -5 },   // 0x42 'B'
  {    98,   5,   6,   6,    1,   -5 },   // 0x43 'C'
  {   102,   5,   6,   6,    1,   -5 },   // 0x44 'D'
  {   106,   3,   6,   4,    1,   -5 },   // 0x45 'E'
  {   109,   3,   6,   4,    1,   -5 },   // 0x46 'F'
  {   112,   6,   6,   6,    1,   -5 },   // 0x47 'G'
  {   117,   5,   6,   6,    1,   -5 },   // 0x48 'H'
  {   121,   1,   6,   3,    1,   -5 },   // 0x49 'I'
  {   122,   3,   6,   3,    0,   -5 },   // 0x4A 'J'
  {   125,   5,   6,   5,    1,   -5 },   // 0x4B 'K'
  {   129,   3,   6,   4,    1,   -5 },   // 0x4C 'L'
  {   132,   7,   6,   7,    0,   -5 },   // 0x4D 'M'
  {   138,   5,   6,   7,    1,   -5 },   // 0x4E 'N'
  {   142,   6,   6,   7,    1,   -5 },   // 0x4F 'O'
  {   147,   4,   6,   4,    1,   -5 },   // 0x50 'P'
  {   150,   6,   7,   7,    1,   -5 },   // 0x51 'Q'
  {   156,   4,   6,   5,    1,   -5 },   // 0x52 'R'
  {   159,   4,   6,   4,    1,   -5 },   // 0x53 'S'
  {   162,   3,   6,   4,    1,   -5 },   // 0x54 'T'
  {   165,   5,   6,   6,    1,   -5 },   // 0x55 'U'
  {   169,   5,   6,   5,    0,   -5 },   // 0x56 'V'
  {   173,   8,   6,   8,    0,   -5 },   // 0x57 'W'
  {   179,   5,   6,   5,    0,   -5 },   // 0x58 'X'
  {   183,   5,   6,   5,    0,   -5 },   // 0x59 'Y'
  {   187,   5,   6,   5,    0,   -5 },   // 0x5A 'Z'
  {   191,   2,   7,   3,    1,   -5 },   // 0x5B '['
  {   193,   4,   6,   4,    0,   -5 },   // 0x5C '\'
  {   196,   2,   7,   3,    1,   -5 },   // 0x5D ']'
  {   198,   5,   2,   8,    2,   -5 },   // 0x5E '^'
  {   200,   4,   1,   4,    0,    2 },   // 0x5F '_'
  {   201,   2,   1,   4,    1,   -4 },   // 0x60 '`'
  {   202,   4,   4,   5,    1,   -3 },   // 0x61 'a'
  {   204,   4,   6,   5,    1,   -5 },   // 0x62 'b'
  {   207,   3,   4,   4,    1,   -3 },   // 0x63 'c'
  {   209,   4,   6,   5,    1,   -5 },   // 0x64 'd'
  {   212,   4,   4,   4,    1,   -3 },   // 0x65 'e'
  {   214,   2,   6,   3,    1,   -5 },   // 0x66 'f'
  {   216,   4,   6,   5,    1,   -3 },   // 0x67 'g'
  {   219,   4,   6,   5,    1,   -5 },   // 0x68 'h'
  {   222,   1,   6,   2,    1,   -5 },   // 0x69 'i'
  {   223,   1,   8,   2,    1,   -5 },   // 0x6A 'j'
  {   224,   4,   6,   4,    1,   -5 },   // 0x6B 'k'
  {   227,   1,   6,   2,    1,   -5 },   // 0x6C 'l'
  {   228,   7,   4,   7,    1,   -3 },   // 0x6D 'm'
  {   232,   4,   4,   5,    1,   -3 },   // 0x6E 'n'
  {   234,   4,   4,   5,    1,   -3 },   // 0x6F 'o'
  {   236,   4,   6,   5,    1,   -3 },   // 0x70 'p'
  {   239,   4,   6,   5,    1,   -3 },   // 0x71 'q'
  {   242,   3,   4,   3,    1,   -3 },   // 0x72 'r'
  {   244,   3,   4,   4,    1,   -3 },   // 0x73 's'
  {   246,   3,   5,   2,    0,   -4 },   // 0x74 't'
  {   248,   4,   4,   5,    1,   -3 },   // 0x75 'u'
  {   250,   5,   4,   4,    0,   -3 },   // 0x76 'v'
  {   253,   6,   4,   6,    0,   -3 },   // 0x77 'w'
  {   256,   4,   4,   4,    0,   -3 },   // 0x78 'x'
  {   258,   4,   6,   4,    0,   -3 },   // 0x79 'y'
  {   261,   4,   4,   4,    0,   -3 },   // 0x7A 'z'
  {   263,   3,   7,   4,    1,   -5 },   // 0x7B '{'
  {   266,   1,   8,   4,    2,   -5 },   // 0x7C '|'
  {   267,   3,   7,   4,    0,   -5 },   // 0x7D '}'
  {   270,   5,   2,   7,    1,   -3 } }; // 0x7E '~'

const GFXfont Futura_Heavy_font4pt7b PROGMEM = {
  (uint8_t  *)Futura_Heavy_font4pt7bBitmaps,
  (GFXglyph *)Futura_Heavy_font4pt7bGlyphs,
  0x20, 0x7E, 9 };

// Approx. 944 bytes

**Comment by Galaxy-Man** [comment, marked as answer]
Resolved.

https://tchapi.github.io/Adafruit-GFX-Font-Customiser/

**Reply by sbrin** [reply]
Hey, thanks for that link. I was looking for solution to get BDF file, but this customiser doesn't have such feature. How did you solve your original issue?
